internal/core/legacy/dao/application.go

```diff
@@ -131,7 +131,7 @@ func (client *DBClient) GetApplicationsByIDs(orgID *int64, projectID *int64, app
 		db = db.Order(fmt.Sprintf("%s", request.OrderBy))
 	}
 	if request.Query != "" {
-		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%"))
+		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%")).Order(gorm.Expr("CHAR_LENGTH(display_name)")).Order(gorm.Expr("CHAR_LENGTH(name)"))
```

Comment:
Test `full text` index.

Reply:
Full-text index is a little cumbersome compared to simple matching and requires additional memory consumption. This field is quite short, making a direct `like` search and sort by their char_length and the index of the first occurrence more suitable.

Reply:
However, use `gorm.Expr()` in mysql to sort the records can't use sqlite3 to write unit-test to test it, the function be used is only in mysql.